### ISSUE TITLE
Upgrade from 8.0.0 to 8.0.1 via CentOS 7 RPMs: unable to activate plugins

### ISSUE THREAD

**Issue by ghost** [owncloud/core#14708]
I just upgraded from 8.0.0 to 0:8.0.1-12.1 and now all plugins are disabled as well as can't be enabled.

![sshot](https://cloud.githubusercontent.com/assets/312549/6505767/b15650cc-c345-11e4-88f7-e7b649f1c70f.png)

After clicking on `Apps -> Recommended -> Contacts 8.0` I am left with the error message:

```
App does not provide an info.xml file
```

This happens for all plugins in the `Recommended` section.

As far as I can tell the `info.xml` files are there:

```
# find /var/www/html/owncloud/ -name info.xml
/var/www/html/owncloud/apps/external/appinfo/info.xml
/var/www/html/owncloud/apps/user_external/appinfo/info.xml
/var/www/html/owncloud/apps/gallery/appinfo/info.xml
/var/www/html/owncloud/apps/files/appinfo/info.xml
/var/www/html/owncloud/apps/provisioning_api/appinfo/info.xml
/var/www/html/owncloud/apps/firstrunwizard/appinfo/info.xml
/var/www/html/owncloud/apps/templateeditor/appinfo/info.xml
/var/www/html/owncloud/apps/files_texteditor/appinfo/info.xml
/var/www/html/owncloud/apps/files_pdfviewer/appinfo/info.xml
/var/www/html/owncloud/apps/files_external/appinfo/info.xml
/var/www/html/owncloud/apps/files_videoviewer/appinfo/info.xml
/var/www/html/owncloud/apps/files_trashbin/appinfo/info.xml
/var/www/html/owncloud/apps/files_encryption/appinfo/info.xml
/var/www/html/owncloud/apps/files_locking/appinfo/info.xml
/var/www/html/owncloud/apps/files_versions/appinfo/info.xml
/var/www/html/owncloud/apps/activity/appinfo/info.xml
/var/www/html/owncloud/apps/user_ldap/appinfo/info.xml
/var/www/html/owncloud/apps/updater/appinfo/info.xml
/var/www/html/owncloud/apps/user_webdavauth/appinfo/info.xml
/var/www/html/owncloud/apps/files_sharing/appinfo/info.xml
/var/www/html/owncloud/apps/search_lucene/appinfo/info.xml
/var/www/html/owncloud/apps/contacts/appinfo/info.xml
/var/www/html/owncloud/apps/calendar/appinfo/info.xml
/var/www/html/owncloud/apps/documents/appinfo/info.xml
```

**Comment by karlitschek** [comment]
can you give us some more information about you system. Do you have zip support installed. Are you able to install anything from the appstore?

**Comment by DeepDiver1975** [comment]
> /var/www/html/owncloud/apps/contacts/appinfo/info.xml

@mrothe looks like there is already an older version of the contacts app installed - please move that outside of the owncloud installation - thx

**Comment by mrothe** [comment]
[ I confused upgrade and update previously. I'll use "update" from now on to describe a change from 8.0.0 to 8.0.1. ]

@karlitschek I do have `unzip` installed if that is what you mean by "zip support".

@DeepDiver1975 I now used the "Uninstall app" links in `Apps -> Not enabled`. The old versions are gone now:

```
# find /var/www/html/owncloud/ -name info.xml|sort
/var/www/html/owncloud/apps/activity/appinfo/info.xml
/var/www/html/owncloud/apps/external/appinfo/info.xml
/var/www/html/owncloud/apps/files/appinfo/info.xml
/var/www/html/owncloud/apps/files_encryption/appinfo/info.xml
/var/www/html/owncloud/apps/files_external/appinfo/info.xml
/var/www/html/owncloud/apps/files_locking/appinfo/info.xml
/var/www/html/owncloud/apps/files_pdfviewer/appinfo/info.xml
/var/www/html/owncloud/apps/files_sharing/appinfo/info.xml
/var/www/html/owncloud/apps/files_texteditor/appinfo/info.xml
/var/www/html/owncloud/apps/files_trashbin/appinfo/info.xml
/var/www/html/owncloud/apps/files_versions/appinfo/info.xml
/var/www/html/owncloud/apps/files_videoviewer/appinfo/info.xml
/var/www/html/owncloud/apps/firstrunwizard/appinfo/info.xml
/var/www/html/owncloud/apps/gallery/appinfo/info.xml
/var/www/html/owncloud/apps/provisioning_api/appinfo/info.xml
/var/www/html/owncloud/apps/templateeditor/appinfo/info.xml
/var/www/html/owncloud/apps/updater/appinfo/info.xml
/var/www/html/owncloud/apps/user_external/appinfo/info.xml
/var/www/html/owncloud/apps/user_ldap/appinfo/info.xml
/var/www/html/owncloud/apps/user_webdavauth/appinfo/info.xml
```

I still get the same error message.

From the [docs](https://doc.owncloud.org/server/8.0/admin_manual/maintenance/update.html) it is not obvious that a simple `yum update` wouldn't do the trick of updating ownCloud.

I am fairly new to ownCloud so I am not sure what I am missing, sorry.

**Comment by mrothe** [comment]
@karlitschek I did not try installing apps from the app store. I am not sure how I'd do that. I tried enabling the "Imprint" App though, which was previously never enabled in this particular installation.

**Comment by PVince81** [comment]
@mrothe did you install the RPM for calendar/contacts too ? (they are separate packages now)

@jnweiger if those aren't installed, shouldn't the calendar and contacts folders be automatically removed by rpm during upgrade ?

**Comment by mrothe** [comment]
A fresh install does not show this behavior, i.e., I can click on "Enable" and "Contacts 8.0" is enabled.

@PVince81 which package would calendar/contacts be in? I have `owncloud-server` and its dependencies installed.

**Comment by PVince81** [comment]
Can you post a list of all installed "owncloud-*" packages and their relevant versions ? They should all be on 8.0.1 after the upgrade. Maybe some of them didn't update properly ?

**Comment by PVince81** [comment]
(not updating properly would mean that the package "requires" field wasn't set properly)

**Comment by mrothe** [comment]
All versions are up-to-date:

```
# yum list owncloud-*
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: ftp.halifax.rwth-aachen.de
 * epel: vesta.informatik.rwth-aachen.de
 * extras: ftp.plusline.de
 * updates: ftp.halifax.rwth-aachen.de
Installed Packages
owncloud-3rdparty.noarch                    8.0.1-12.1         @owncloud
owncloud-app-activity.noarch                8.0.1-12.1         @owncloud
owncloud-app-external.noarch                8.0.1-12.1         @owncloud
owncloud-app-files.noarch                   8.0.1-12.1         @owncloud
owncloud-app-files_encryption.noarch        8.0.1-12.1         @owncloud
owncloud-app-files_external.noarch          8.0.1-12.1         @owncloud
owncloud-app-files_locking.noarch           8.0.1-12.1         @owncloud
owncloud-app-files_pdfviewer.noarch         8.0.1-12.1         @owncloud
owncloud-app-files_sharing.noarch           8.0.1-12.1         @owncloud
owncloud-app-files_texteditor.noarch        8.0.1-12.1         @owncloud
owncloud-app-files_trashbin.noarch          8.0.1-12.1         @owncloud
owncloud-app-files_versions.noarch          8.0.1-12.1         @owncloud
owncloud-app-files_videoviewer.noarch       8.0.1-12.1         @owncloud
owncloud-app-firstrunwizard.noarch          8.0.1-12.1         @owncloud
owncloud-app-gallery.noarch                 8.0.1-12.1         @owncloud
owncloud-app-provisioning_api.noarch        8.0.1-12.1         @owncloud
owncloud-app-templateeditor.noarch          8.0.1-12.1         @owncloud
owncloud-app-updater.noarch                 8.0.1-12.1         @owncloud
owncloud-app-user_external.noarch           8.0.1-12.1         @owncloud
owncloud-app-user_ldap.noarch               8.0.1-12.1         @owncloud
owncloud-app-user_webdavauth.noarch         8.0.1-12.1         @owncloud
owncloud-server.noarch                      8.0.1-12.1         @owncloud
Available Packages
owncloud.noarch                             8.0.1-12.1         owncloud 
owncloud-config-apache.noarch               8.0.1-12.1         owncloud 
owncloud-httpd.noarch                       7.0.4-2.el7        epel     
owncloud-mysql.noarch                       7.0.4-2.el7        epel     
owncloud-nginx.noarch                       7.0.4-2.el7        epel     
owncloud-postgresql.noarch                  7.0.4-2.el7        epel     
owncloud-server-scl-php54.noarch            8.0.1-12.1         owncloud 
owncloud-sqlite.noarch                      7.0.4-2.el7        epel
```

**Comment by PVince81** [comment]
Ok thanks.
Forget about the owncloud-calendar and owncloud-contact packages. There are no such package, I was just confused. Sorry.

**Comment by PVince81** [comment]
To answer your question, calendar/contacts are NOT in any package.

This means that when you had 8.0.0 you had to install them from the apps page, which you did.
Then upon upgrading to 8.0.1, the OC code and core apps were upgraded through the RPM packages, but the calendar/contact apps were left on the old version (obviously) and also auto-disabled.

Then you tried to reenable them in the apps page and that's when it went wrong. It should actually update the app if the code is already there. But yours failed even earlier.

@karlitschek @georgehrke that info.xml message looks like the OCSID issue ?

**Comment by PVince81** [comment]
also the button is supposed to say "Update", not "Enable". Unless you took the screenshot after uninstalling the app ?

**Comment by PVince81** [comment]
@karlitscheck @georgehrke it seems there are two contact apps in the apps page.

If you click "Recommended" it will show "Contacts 8.0" and "Calendar 8.0", which is likely to be the ones for 8.0.1 ?!

If you click "Not enabled" in the sidebar, you'll get another version called "Contacts" and "Calendar".
Strangely I cannot find "Contacts 8.0" or "Calendar 8.0" on the "Not enabled" list.

Are these the same entries with same OCID ?

**Comment by PVince81** [comment]
Try these steps:
1. Setup OC 8.0.0
2. Install calendar and contacts from the apps page (Recommended section)
3. Upgrade to OC 8.0.1, notice that both apps get disabled
4. Go to apps page, recommended section
5. The button says "Enable" instead of "Upgrade".

**Comment by georgehrke** [comment]
> 5. The button says "Enable" instead of "Upgrade". 

Because Calendar 8.0 is a completely independent app and doesn't know, that another app called calendar exists. It's like installing a new app

**Comment by PVince81** [comment]
OK so we'll need #14716 for that

**Comment by PVince81** [comment]
Closing as duplicate of https://github.com/owncloud/core/issues/14711 which provides a fix

**Comment by mrothe** [comment]
Is this really a duplicate? As per my first [comment](https://github.com/owncloud/core/issues/14708#issuecomment-77401642), the directories are gone, but I am still getting the same error message.

**Comment by mrothe** [comment]
I just found the following error in the log:

```
App does not provide an info.xml file
ZipArchive::extractTo(): Invalid or unitialized Zip object at /var/www/html/owncloud/lib/private/archive/zip.php#135
```

**Comment by mrothe** [comment]
This is php verison 5.4.16 if that matters. It is the stable version of CentOS 7.

**Comment by PVince81** [comment]
@mrothe you said you used "uninstall app" which would delete the folder.

Let's reopen to investigate this case.

**Comment by PVince81** [comment]
Or like you said it could be the zip issue.
Because normally the error message is "App folder already exists" but in your case it is "App doesn't contain info.xml" which means it is likely that it could not extract the zip/tar properly for some reason

**Comment by PVince81** [comment]
@georgehrke @VicDeo does that zip error message rings any bells ?

**Comment by mrothe** [comment]
I was saying that I am not able to reproduce this with a new install in a previous (comment)[https://github.com/owncloud/core/issues/14708#issuecomment-77411434], but I did not pay attention to the ownCloud versions involved.

It seems you withdrew version 8.0.1 from the stable repository. My test setup has version 8.0.0 installed, because it uses the stable CentOS 7 repo.

In other words: I did *not* show that this does *not* happen with a new installation.

**Comment by VicDeo** [comment]
Sounds like https://github.com/owncloud/core/issues/14226#issuecomment-76234643

**Comment by mscheutz** [comment]
Actually, I had the same problem getting "App does not provide an info.xml file" from one day to the next without having changed anything and before the Calendar app had worked just fine, but after (presumably some update to owncloud) it stopped working.  Looking into it, I noticed that the xml format had changed in info.xml and that this change seems to be causing the "Invalid or unitialized Zip object" error.  Specifically, when I renamed the old <require> tag to what seems to be the new tag <requiremin>, the Calendar app starting working again (i.e., I could enable the Calendar without getting the error)

**Comment by LukasReschke** [comment]
Do you all have the PHP cURL extension installed?

**Comment by mrothe** [comment]
@LukasReschke Do you mean the [PEAR Net_Curl](http://pear.php.net/package/Net_Curl) package (`php-pear-Net-Curl` on CentOS)? I do *not* have that installed.

**Comment by DeepDiver1975** [comment]
@mrothe not the pear package but the php extension. I have no idea how Centos is packaging php - maybe php-curl?

You might want to paste the result of 

```sh
$ php --info

**Comment by mrothe** [comment]
Here is the relevant part of the output of `php --info`. So, yes, curl support is enabled.

```
phpinfo()
PHP Version => 5.4.16

[...]

curl

cURL support => enabled
cURL Information => 7.29.0
Age => 3
Features
AsynchDNS => Yes
Debug => No
GSS-Negotiate => Yes
IDN => Yes
IPv6 => Yes
Largefile => Yes
NTLM => Yes
SPNEGO => No
SSL => Yes
SSPI => No
krb4 => No
libz => Yes
CharConv => No
Protocols => dict, file, ftp, ftps, gopher, http, https, imap, imaps, ldap, ldaps, pop3, pop3s, rtsp, scp, sftp, smtp, smtps, telnet, tftp
Host => x86_64-redhat-linux-gnu
SSL Version => NSS/3.15.4
ZLib Version => 1.2.7
libSSH Version => libssh2/1.4.3

[...]
```

On CentOS it seems to be part of php:

```
# repoquery -l php-common|grep curl
/etc/php.d/curl.ini
/usr/lib64/php/modules/curl.so
```

**Comment by mscheutz** [comment]
I did install it,  but it was only part of the problem


From my Android phone on T-Mobile. The first nationwide 4G network.

<div>-------- Original message --------</div><div>From: Lukas Reschke <notifications@github.com> </div><div>Date:03/12/2015  1:35 PM  (GMT-05:00) </div><div>To: owncloud/core <core@noreply.github.com> </div><div>Cc: mscheutz <matthias.scheutz@tufts.edu> </div><div>Subject: Re: [core] Upgrade from 8.0.0 to 8.0.1 via CentOS 7 RPMs: unable to
  activate plugins (#14708) </div><div>
</div>Do you all have the PHP cURL extension installed?

**Comment by karlitschek** [comment]
@DeepDiver1975 @LukasReschke isn´t this the stuff that got fixed lately?

**Comment by EvgeniySpinov** [comment]
I'm getting the same issue on CentOS 6, Owncloud 8.0.2. Have all packages installed including referred issues here.

I've made a clean install from site. 

Maybe I can provide more information in order to resolve issue?

**Comment by LukasReschke** [comment]
Ewww… I really like CentOS and RHEL… "Enterprise-ready", preparing PR. @EvgeniySpinov I will mention you latter with a code change that you could test. That would be superb.

**Comment by EvgeniySpinov** [comment]
Em, #14840 solved issue for me. Looks like self-signed certificates will create issues for new users. Probably there is way to show warning/automatically process situation when user has self-signed certificate?

**Comment by LukasReschke** [comment]
> Em, #14840 solved issue for me. Looks like self-signed certificates will create issues for new users. Probably there is way to show warning/automatically process situation when user has self-signed certificate?

That's not a fix - just a workaround :-)

**Comment by EvgeniySpinov** [comment]
Yep, never said differently, I will remove that after installing signed certificate. However it's hard to find solution as there are too many issues like this with same sympthons, i.e. log entries, but with different solution.

Just let me know if I can be in use of fixing this issue.

**Comment by PVince81** [comment]
Are you guys still having these issues with 8.0.5 or 8.1 ?

**Comment by mscheutz** [comment]
seems to work for us now
On Mon, 2015-08-03 at 08:30 -0700, Vincent Petry wrote:
> Are you guys still having these issues with 8.0.5 or 8.1 ?
> —
> Reply to this email directly or view it on GitHub.
-- 
Matthias Scheutz, Ph.D. Ph.D.
-------------------------------------------------
Professor, Cognitive Science and Computer Science
Bernard M. Gordon Senior Faculty Fellow
Director, Human-Robot Interaction Laboratory
Department of Computer Science
Tufts University, Medford, MA 02155, USA

**Comment by PVince81** [comment]
Thanks, closing.

@EvgeniySpinov I hope the problem is solved for you too in 8.1.3. If not, feel free to reopen.
